app/models/user/academics/student_group.rb

```diff
@@ -0,0 +1,21 @@
+module User
+  module Academics
+    class StudentGroup
+      def initialize(data)
+        @data = data
+      end
+
+      def code
+        @data['student_group_code']
+      end
+
+      def as_json(options={})
+        {
+          code: @data['student_group_code'],
+          description: @data['student_group_description'],
+          from_date: @data['from_date']&.to_date,
```

Comment:
You could make your JSON interface easier to deal with on the client side by making the to_json output `fromDate`. I know from experience 🤦‍♂️

Reply:
@developish  - We actually have a `HashConverter` module that we can use to recursively convert underscore case keys to camelcase for the benefit of the front-end. This has been an ongoing debate amongst developers concerning which level to do this.

Within `as_json`, it makes sense to do it there.